monkey/monkey_island/cc/repository/i_credentials_repository.py

```diff
@@ -0,0 +1,75 @@
+from abc import ABC
+from typing import Any, Sequence
+
+from monkey.cc.models import ConfiguredCredentials, StolenCredentials
+
+
+class ICredentialsRepository(ABC):
+    def get_configured_credentials(self) -> Sequence[ConfiguredCredentials]:
+        """
+        Retrieves all credentials that were configured.
+
+        :raises RetrievalError: if it can't get the credentials
+        :return: Sequence of configured credentials
+        """
+        pass
+
+    def save_configured_credentials(self, configured_credentials: ConfiguredCredentials):
+        """
+        Saves credentials which are configured.
+
+        :param configured_credentials: Credentials that are going to be stored.
+        :raises StorageError: if the credentials can't be saved.
+        """
+        pass
+
+    def remove_configured_credentials(self):
+        """
+        It removes all configured credentials.
+
+        :raises RemovalError: if the configured credentials can't be removed.
+        """
+        pass
+
+    def get_stolen_credentials(self) -> Sequence[StolenCredentials]:
+        """
+        Retrieves credentials that are stolen
+
+        :raises RetrievalError: if it can't get the credentials
+        :return: Sequence of all stolen credentials
+        """
+        pass
+
+    def save_stolen_credentials(self, stolen_credentials: StolenCredentials):
+        """
+        Saves credentials which are stolen.
+
+        :param stolen_credentials: Credentials that are going to be stored.
+        :raises StorageError: if the credentials can't be saved.
+        """
+        pass
+
+    def remove_stolen_credentials(self):
+        """
+        It removes all credentials from the repository.
+
+        :raises RemovalError: if the credentials can't be removed.
+        """
+        pass
+
+    def get_all_credentials(self) -> Sequence[Any[StolenCredentials, ConfiguredCredentials]]:
+        """
+        It retrieves stolen and configured credentials.
+
+        :raises RetrievalError: if it can't get the credentials
```

Comment:
Use the passive tense:
```suggestion
        :raises RetrievalError: If the credentials can't be retrieved
```

We can also be a bit more specific:
```suggestion
        :raises RetrievalError: If an error is encountered while attempting to retrieve the credentials
```